docker/Dockerfile

```diff
@@ -20,30 +23,33 @@ RUN git clone https://github.com/gleitz/npiet /opt/npiet
 # npiet includes `sys/malloc.h` which doesn't exist. Malloc definitions come
 # from stdlib.h.
 RUN sed -i 's|sys/malloc\.h|stdlib.h|g' /opt/npiet/npiet-foogol.*
-RUN cd /opt/npiet && ./configure && make && make install
+WORKDIR /opt/npiet
+RUN ./configure && make && make install
 
 # Download jsteg
 RUN wget -O /usr/local/bin/jsteg https://github.com/lukechampine/jsteg/releases/download/v0.3.0/jsteg-linux-amd64 && chmod +x /usr/local/bin/jsteg
 RUN wget -O /usr/local/bin/slink https://github.com/lukechampine/jsteg/releases/download/v0.3.0/slink-linux-amd64 && chmod +x /usr/local/bin/slink
 
 # Download, compile and install snow
 RUN wget -O /usr/snow.tar.gz http://www.darkside.com.au/snow/snow-20130616.tar.gz
-RUN cd /usr && tar -xvf snow.tar.gz
-RUN cd /usr/snow-20130616/ && make 
+WORKDIR /usr
+RUN tar -xvf snow.tar.gz
+WORKDIR /usr/snow-20130616/
+RUN make
 RUN cp /usr/snow-20130616/snow /usr/local/bin/snow && chmod +x /usr/local/bin/snow
 
 # Clone Katana Repository
 RUN git clone --recursive https://github.com/JohnHammond/katana.git /katana
```

Comment:
I know your intention on this but in case of docker best practices you shouldn't reclone the sources.